common-tooling/action.yml

```diff
@@ -0,0 +1,302 @@
+    default: 'tonistiigi/binfmt:latest'
+    required: false
+  qemu-platforms:
+    description: 'Platforms to install (e.g. arm64,riscv64,arm)'
+    default: 'all'
+    required: false
+  qemu-enabled:
+    description: 'Whether to install qemu or not'
+    default: 'true'
+  # Java / Maven
+  java-version:
+    description: 'The Java version to set up'
+    default: '17'
+    required: false
+  java-distribution:
+    description: 'Java distribution'
+    default: 'temurin'
+    required: false
+  java-cache-prefix:
+    description: 'Cache key prefix'
+    required: false
+  java-cache-path:
+    description: 'Cache path'
+    default: '~/.m2/repository'
+    required: false
+  java-cache-path-add:
+    description: 'Additional item for cache path'
+    required: false
+  java-maven-version:
+    description: 'The Maven version to set up'
+    default: '3.9.2'
+    required: false
+  java-enabled:
+    description: 'Whether to install java or not'
+    default: 'true'
+  # Python
+  python-version:
+    description: "Version range or exact version of Python or PyPy to use, using SemVer's version range syntax. Reads from .python-version if unset."
+    default: "3.11"
+  python-version-file:
+    description: "File containing the Python version to use. Example: .python-version"
+  python-cache:
+    description: "Used to specify a package manager for caching in the default directory. Supported values: pip, pipenv, poetry."
+    required: false
+  python-architecture:
+    description: "The target architecture (x86, x64) of the Python or PyPy interpreter."
+  python-check-latest:
+    description: "Set this option if you want the action to check for the latest available version that satisfies the version spec."
+    default: "false"
+  python-token:
+    description: "The token used to authenticate when fetching Python distributions from https://github.com/actions/python-versions. When running this action on github.com, the default value is sufficient. When running on GHES, you can pass a personal access token for github.com if you are experiencing rate limiting."
+    default: ${{ github.server_url == 'https://github.com' && github.token || '' }}
+  python-cache-dependency-path:
+    description: "Used to specify the path to dependency files. Supports wildcards or a list of file names for caching multiple dependencies."
+  python-update-environment:
+    description: "Set this option if you want the action to update environment variables."
+    default: 'true'
+  python-allow-prereleases:
+    description: "When 'true', a version range passed to 'python-version' input will match prerelease versions if no GA versions are found. Only 'x.y' version range is supported for CPython."
+    default: "false"
+  python-enabled:
+    description: 'Whether to install python or not'
+    default: 'true'
+  # misc
+  overwrite:
+    description: 'Defines whether on hosted runners the present version should be overwritten'
+    default: 'true'
```

Comment:
should it rather be `true` so the default behaviour in SaaS runners takes less runtime?

Reply:
the idea was of a drop-in replacement to be equal for either hosted or self-hosted. Meaning if you run it on either, you should expect to get the same results but I'm fine changing that up.

Reply:
I'd see that as an action that makes SH behave the same as SaaS (i.e. on SaaS it's a no-op that installs nothing and on SH it installs the tooling missing there to match default `ubuntu-latest` SaaS one)

Reply:
switched to `false` as default.